challengeA/yoshikawa_keishi/lib/simulator.rb

```diff
@@ -0,0 +1,102 @@
+class Simulator 
+  attr_reader :amp, :usage
+  @@EP_planB_SET = {rate: {0 => 19.88, 120 => 26.48, 300 => 30.57}, basicChargeRate: 28.6 }
+  @@TG_ZUTTOMO_SET = {rate: {0 => 23.67, 140 => 23.88, 350 => 26.41}, basicChargeRate: 28.6 }
+  @@LOOOP_HOMEPLAN_SET = {basicChargeRate: 26.40 }
+  @@acceptableAmperes = [10, 15, 20, 30, 40, 50, 60]
```

Comment:
東京ガスのずっとも電気1は、30,40,50,60Aが対象なんですよね。
acceptableAmpereはプランごとに持っても良い気がします。
そしてプランごとに定義したacceptableAmpereの和集合が、ユーザーが本Simulatorで診断可能なacceptableAmpereになるかと思います。

Reply:
ご指摘の通りに実装しました。acceptableAmpereの和集合に関しては、外部から変更する必要がないのでクラス変数はやめてグローバル変数にしてみました。理由は、もしコンソールからアンペアを入力する場合にクラスの外でもバリデーションをかけられるからです。

```suggestion
    @@EP_planB_SET = {amperes: [10, 15, 20, 30, 40, 50, 60], rate: {0 => 19.88, 120 => 26.48, 300 => 30.57}, basicChargeRate: 28.6 }
  @@TG_ZUTTOMO_SET = {amperes: [30, 40, 50, 60], rate: {0 => 23.67, 140 => 23.88, 350 => 26.41}, basicChargeRate: 28.6 }
  @@LOOOP_HOMEPLAN_SET = {amperes: [10, 15, 20, 30, 40, 50, 60], rate: {0 => 26.40}, basicChargeRate: 0 }
  $acceptableAmperes = @@EP_planB_SET[:amperes] || @@TG_ZUTTOMO_SET [:amperes] || @@LOOOP_HOMEPLAN_SET[:amperes]
```